audioipc/src/lib.rs

```diff
@@ -140,21 +141,26 @@ impl PlatformHandle {
     }
 
     #[cfg(windows)]
-    pub unsafe fn into_file(self) -> std::fs::File {
-        std::fs::File::from_raw_handle(self.0)
+    pub unsafe fn into_file(&self) -> std::fs::File {
+        std::fs::File::from_raw_handle(self.as_raw())
     }
 
     #[cfg(unix)]
-    pub unsafe fn into_file(self) -> std::fs::File {
-        std::fs::File::from_raw_fd(self.0)
+    pub unsafe fn into_file(&self) -> std::fs::File {
+        std::fs::File::from_raw_fd(self.as_raw())
     }
 
-    pub fn as_raw(self) -> PlatformHandleType {
-        self.0
+    pub unsafe fn as_raw(&self) -> PlatformHandleType {
```

Comment:
Oh, is calling `take()` the reason the function names stay with `into*` or `as*`? Perhaps there are better names ...

Reply:
I think `into` fits better since the source PlatformHandle becomes invalid.  Ideally PlatformHandle's `into_*` functions would take `self` instead, but that'll require significant changes to the API users.

Reply:
I'll rename `as_raw` to `into_raw` and leave `into_file` as-is for now.

Reply:
I'm using unsafe on `to_raw` to indicate the caller needs to take care not to leak the raw handles.